plugins/lsp/conform-nvim.nix

```diff
@@ -0,0 +1,119 @@
+                javascript = [ [ "prettierd" "prettier" ] ];
+                # Use the "*" filetype to run formatters on all filetypes.
+                "*" = [ "codespell" ];
+                # Use the "_" filetype to run formatters on filetypes that don't
+                # have other formatters configured.
+                "_" = [ "trim_whitespace" ];
+               };
+          ```
+        '';
+
+      formatOnSave =
+        helpers.defaultNullOpts.mkNullable (types.submodule {
+          options = {
+            lspFallback = mkOption {
+              type = types.bool;
+              default = true;
+              description = "See :help conform.format for details.";
+            };
+            timeoutMs = mkOption {
+              type = types.int;
+              default = 500;
+              description = "See :help conform.format for details.";
+            };
+          };
+        })
+        "see documentation"
+        ''
+          If this is set, Conform will run the formatter on save.
+          It will pass the table to conform.format().
+          This can also be a function that returns the table.
+          See :help conform.format for details.
+        '';
+
+      formatAfterSave =
+        helpers.defaultNullOpts.mkNullable (types.submodule {
+          options = {
+            lspFallback = mkOption {
+              type = types.bool;
+              default = true;
+              description = "See :help conform.format for details.";
+            };
+          };
+        })
+        "see documentation"
+        ''
+          If this is set, Conform will run the formatter asynchronously after save.
+          It will pass the table to conform.format().
+          This can also be a function that returns the table.
+        '';
+
+      logLevel =
+        helpers.defaultNullOpts.mkEnumFirstDefault
+        ["ERROR" "DEBUG" "INFO" "TRACE" "WARN" "OFF"]
+        "Set the log level. Use `:ConformInfo` to see the location of the log file.";
+
+      notifyOnError =
+        helpers.defaultNullOpts.mkBool true
+        "Conform will notify you when a formatter errors";
+
+      formatters =
+        helpers.defaultNullOpts.mkNullable (types.attrsOf types.anything)
+        "see documentation"
+        "Custom formatters and changes to built-in formatters";
+    };
+
+  config = let
+    setupOptions = with cfg;
+      {
+        formatters_by_ft = helpers.ifNonNull' cfg.formattersByFt;
+        format_on_save = helpers.ifNonNull' cfg.formatOnSave {
+          lsp_fallback = cfg.formatOnSave.lspFallback;
+          timeout_ms = cfg.formatOnSave.timeoutMs;
+        };
+        format_after_save = helpers.ifNonNull' cfg.formatOnSave {
+          lsp_fallback = cfg.formatOnSave.lspFallback;
+        };
+        log_level = helpers.ifNonNull' cfg.logLevel (helpers.mkRaw "vim.log.levels.${cfg.logLevel}");
+        notify_on_error = helpers.ifNonNull' cfg.notifyOnError;
+        formatters = helpers.ifNonNull' cfg.formatters;
```

Comment:
You don't need helpers.ifNonNull' if you don't do something with the value in a simple assign like this. (same for formatters_by_ft)